shacl12-node-expr/index.html

```diff
@@ -813,12 +814,34 @@ <h3>Named Parameter Functions</h3>
 						the <a href="#EstonianCompanyShapeExample">Estonian Company Shape example</a>.
 					</p>
 				</section>
+
 				<section id="ListParameterFunction">
 					<h3>List Parameter Functions</h3>
 					<p>
 						A <dfn>list parameter function</dfn> is a <a>node expression function</a>
 						that is represented by a <a>blank node</a> that is the <a>subject</a> of a single
-						<a>triple</a> where the <a>object</a> is a <a>SHACL list</a>.
+						<a>triple</a> where the <a>object</a> <code>o</code> is conforming to one of the following syntax rules, in order:
+					</p>
+					<ol>
+						<li>
+							If <code>o</code> is the empty <a>SHACL list</a> <code>rdf:nil</code> (written as <code>()</code> in Turtle)
+							then the arguments are the empty list.<br />
+							Example: <code>[ sparql:now () ]</code>
+						</li>
+						<li>
+							If <code>o</code> is a <a>blank node</a> that is a <a>well-formed</a> <a>SHACL list</a> where all
+							<a>members</a> are <a>well-formed</a> <a>node expressions</a>,
+							then the arguments are the <a>members</a> of that list.<br />
+							Examples: <code>[ sparql:plus ( 38 4 ) ]</code> and <code>[ sparql:abs ( -42 ) ]</code>
+						</li>
+						<li>
+							If the function accepts exactly one parameter (such as <code>sparql:abs</code>)
```

Comment:
> If the function accepts exactly one parameter

This restricts implementation. It requires knowledge of the arity before determining the legality of an expression. 

1. What is the function is withdrawn? Legal syntax become illegal.

2. You can't write a checker without knowing the function - hard for a UI to do checking in the browser as it needs to know the functions at the server side at time of execution.

They are "List Parameter Functions".

Reply:
I think it's not really a pre-condition "if the function accepts exactly one parameter". Instead it could be "if there is exactly one argument (in the function call)".